server/index.js

```diff
@@ -0,0 +1,205 @@
+require('dotenv').config();
+const request = require('request');
+const http = require('http');
+const httpProxy = require('http-proxy');
+const express = require('express');
+const bodyParser = require('body-parser');
+const bcrypt = require('bcrypt');
+const JWT = require('./jwt');
+
+const app = express();
+
+const proxy = httpProxy.createProxyServer({
+  changeOrigin: true,
+  ignorePath: false
+});
+
+proxy.on('proxyReq', function(proxyReq, req, res, options) {
+  proxyReq.setHeader('authorization', `Bearer ${process.env.AIRTABLE_API_KEY}`);
+});
+
+app.use(bodyParser.json());
+
+app.get('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.post('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.put('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.delete('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.post('/register', (req, res) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  const url = `${process.env.AIRTABLE_ENDPOINT_URL}/v0/${process.env.AIRTABLE_BASE_ID}/${process.env.AIRTABLE_USER_TABLE}?fields%5B%5D=username&filterByFormula=username%3D%22${username}%22`;
```

Comment:
I'd like this to be a bit more semantic for future clarity/extensibility - can we use `querystring` here? Something like...

```javascript
const querystring = require('querystring');
const {
  AIRTABLE_ENDPOINT_URL,
  AIRTABLE_BASE_ID,
  AIRTABLE_USER_TABLE
} = process.env;

const url = `${AIRTABLE_ENDPOINT_URL}/v0/${AIRTABLE_BASE_ID}/${AIRTABLE_USER_TABLE}/?${
  querystring.stringify({
     fields: ["username"],
     filterByFormula: `username="${username}"`
  })
}`;
```

Reply:
Also - perhaps this `url` should be a global. `USERNAME_QUERY_URL`?

Reply:
I can play around with this idea more -- maybe I could create a global function that can takes in a "username" parameter (maybe generalized) to allow for string interpolation, because currently if I define it as a global const right now, i get yelled at for an undefined "${username}"

Reply:
Parameterized makes sense here. Can we try it?

Reply:
The latest commit uses Airtable's Object-to-Query-Param-String [solution](https://github.com/Airtable/airtable.js/blob/master/lib/object_to_query_param_string.js)

I used this as my params object to test
`{
     fields: ["username"],
     filterByFormula: `username="${username}"`
  }`

`querystring.stringify` would produce the following string
`fields=username&filterByFormula=username%3D%22testQueryFieldsTest%22`

while Airtable's implementation would process the fields array differently by producing the following url query string
`fields%5B%5D=username&filterByFormula=username%3D%22testQueryFieldsTest%22`